File: CPE/CPE49/30-UVA10050.py
MAX_DAYS = 3660
T = int(input())

for _ in range(T):
    N = int(input()) # of days
    P = int(input()) # of parties
    H = [int(input()) for _ in range(P)]

    days = [0] * (N + 1)
    for h in H:
        for i in range(h, N + 1, h):
            days[i] = 1
    ans = 0
    for i in range(1, N + 1):
        if i % 7 == 6 or i % 7 == 0:
            continue
        ans += days[i]
    print(ans)

File: CPE/CPE49/30-UVA10050.cpp
#include <bits/stdc++.h>
using namespace std;
const int N = 3660;
#define endl '\n'

int main() {
    ios::sync_with_stdio(false); cin.tie(nullptr); cout.tie(nullptr);
    int t, n, p, h, days[N+1], ans;
    cin >> t;
    while (t--) {
        cin >> n >> p;
        memset(days, 0, sizeof(days));
        for (int i=0; i<p; i++) {
            cin >> h;
            for (int j=h; j<=n; j+=h) {
                days[j] = 1;
            }
        }
        ans = 0;
        for (int i=1; i<=n; i++) {
            if (i % 7 == 6 || i % 7 == 0) continue;
            ans += days[i];
        }
        cout << ans << endl;
    }
    return 0;
}